src/compiler/checker.ts

```diff
@@ -22804,7 +22809,7 @@ namespace ts {
 
             const localOrExportSymbol = getExportSymbolOfValueSymbolIfExported(symbol);
             const sourceSymbol = localOrExportSymbol.flags & SymbolFlags.Alias ? resolveAlias(localOrExportSymbol) : localOrExportSymbol;
-            if (getDeclarationNodeFlagsFromSymbol(sourceSymbol) & NodeFlags.Deprecated && isUncalledFunctionReference(node.parent, sourceSymbol)) {

```

Comment:
don't you need to update the other callers of `isUncalledFunctionReference` too?

Reply:
Other callers of `isUncalledFunctionReference` don't use parent and use the _original node_, only in `checkIdentifier` uses `node.parent`.